Game/MyProject/Game/Helper/HelperManager.cpp

```diff
@@ -0,0 +1,548 @@
+    vertex3.position=points[11];
+    
+    vertices.push_back(vertex1);
+    vertices.push_back(vertex2);
+    vertices.push_back(vertex3);
+    
+    vertex1.position=points[6];
+    vertex2.position=points[2];
+    vertex3.position=points[10];
+    
+    vertices.push_back(vertex1);
+    vertices.push_back(vertex2);
+    vertices.push_back(vertex3);
+    
+    vertex1.position=points[8];
+    vertex2.position=points[6];
+    vertex3.position=points[7];
+    
+    vertices.push_back(vertex1);
+    vertices.push_back(vertex2);
+    vertices.push_back(vertex3);
+    
+    vertex1.position=points[9];
+    vertex2.position=points[8];
+    vertex3.position=points[1];
+    
+    vertices.push_back(vertex1);
+    vertices.push_back(vertex2);
+    vertices.push_back(vertex3);
+    
+    vertex1.position=points[0];
+    vertex2.position=points[11];
+    vertex3.position=points[5];
+    
+    vertices.push_back(vertex1);
+    vertices.push_back(vertex2);
+    vertices.push_back(vertex3);
+    
+    std::vector<IvTNPVertex> vertices2;
+    vertices2=RefineTriangles(vertices, 4);
+    sphereMesh->setVertexBuffer(vertices2, format);
+    
+    std::vector<unsigned int> indices;
+    for(int i=0;i<vertices2.size();i++)
+        indices.push_back(i);
+    
+    sphereMesh->setIndexBuffer(indices);
+
+    return sphereMesh;
+}
+
+//-------------------------------------------------------------------------------
+// @ HelperManager::CreateMeshInstance()
+//-------------------------------------------------------------------------------
+std::shared_ptr<MeshInstance>
+HelperManager::CreateMeshInstance(const std::shared_ptr<Mesh> &mesh,
+                                  const char* shaderName,
+                                  const std::vector<std::string>& uniforms)
+{
+    std::shared_ptr<MeshInstance> meshInstance=std::make_shared<MeshInstance>();
+    meshInstance->setMesh(mesh);
+    meshInstance->setShader(shaderName);
+    meshInstance->addShaderUniforms(uniforms);
+    
+    return meshInstance;
+}
+//------------------------------------------------------------------------------
+//  PRIVATE METHODS
+//-------------------------------------------------------------------------------
+
+//-------------------------------------------------------------------------------
+// @ HelperManager::GetMiddlePoint()
+//-------------------------------------------------------------------------------
+IvVector3
+HelperManager::GetMiddlePoint(IvVector3 p1, IvVector3 p2)
+{
+    IvVector3 mid=IvVector3(    (p1.x+p2.x)/2.0,
+                                (p1.y+p2.y)/2.0,
+                                (p1.z+p2.z)/2.0);
+    
+    return mid;
+}
+//-------------------------------------------------------------------------------
+// @ HelperManager::GetMiddlePoint()
+//-------------------------------------------------------------------------------
+std::vector<IvTNPVertex>
+HelperManager::RefineTriangles(std::vector<IvTNPVertex> vertices,int recursionLevel)
```

Comment:
I'm not so sure about this approach. You could just generate all the vertices upfront in the CreateSphereMesh method and avoid this nested-loop recursion.. but if you fix the other things and it works ok you can keep it..